lib/pure/asyncdispatch.nim

```diff
@@ -1216,6 +1216,16 @@ else:
 
   var gDisp{.threadvar.}: owned PDispatcher ## Global dispatcher
 
+  when defined(nuttx):
+    proc addAtExit(quitProc: proc() {.noconv, cdecl.}) {.
+      importc: "atexit", header: "<stdlib.h>".}
```

Comment:
It was marked "deprecated" and I thought it would go away.

```nim
when not defined(nimPreviewSlimSystem):
  proc addQuitProc*(quitProc: proc() {.noconv.}) {.
    importc: "atexit", header: "<stdlib.h>", deprecated: "use exitprocs.addExitProc".}
```

Reply:
In the exitprocs, there is a "callClosures" that registers with the OS in atexit() and executes functions that the user registers.
Would it be a good idea to change it so that it executes while removing functions from the gFuns array?
(The second time the task is invoked, gFuns.len == 0, so atexit() will be executed again).